### PR TITLE
Initial creation of lexical model submission guide

### LINKED ISSUE #69: Add tutorial how to submit to lexical-models repo
See #35

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #69

Based on the keyboard submission guide

### CODE REVIEW

File: developer/lexical-models/index.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+<h2>Introduction</h2>
+
+<p>
+  The Keyman Cloud repository is the master source for all lexical models distributed within the Keyman mobile apps.
+  The use of these dictionaries started with Keyman 12.
+</p>
+
+<h2>The Keyman Lexical Models Repository</h2>
+<p>
+  We now have a single location for open source lexical models on github.com
```
Comment (mcdurdin) [inline developer/lexical-models/index.php]:
```suggestion
  Keyman can automatically download lexical models that are shared through our open source lexical model repository on github.com:
```

File: developer/lexical-models/submission/index.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  // Required
+  head([
+    'title' =>'Guide: Lexical Model Submissions'
+  ]);
+?>
+
+<h1>Guide for Lexical Model Submissions</h1>
+<p>
+  This is a guide for submitting lexical models to the lexical-models repository.
+  Users will need to be proficient enough with Git to keep their own fork in sync with the upstream branch.
```
Comment (mcdurdin) [inline developer/lexical-models/submission/index.php]:
```suggestion
  Lexical model developers will need to be proficient enough with Git to keep their own fork in sync with the upstream branch.
```

File: developer/lexical-models/submission/index.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+</p>
+
+<p class="hint">
+  <b>Note:</b> This guide is written for Windows users, so other OS users need to adjust paths accordingly.
```
Comment (mcdurdin) [inline developer/lexical-models/submission/index.php]:
```suggestion
  <b>Note:</b> This guide is written for lexical model developers using Windows, so other OS users need to adjust paths accordingly.
```

File: developer/lexical-models/submission/step-0.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    ]);
+?>
+
+<h1 id="get-source">Preliminary Step: Is Your Lexical Model Ready?</h1>
+
+<h2>Folder and File Structure and Naming</h2>
+
+<p>
+    The files for each lexical model are organized in a specific fashion to make it possible for the lexical models to be built automatically.
+    If you are using Keyman Developer (version 12 or later), you can select "New Project" from the "Project" menu, then "Wordlist Lexical Model"
```
Comment (mcdurdin) [inline developer/lexical-models/submission/step-0.php]:
```suggestion
    In Keyman Developer (version 12 or later), you can select "New Project" from the "Project" menu, then "Wordlist Lexical Model"
```

File: developer/lexical-models/_shared/development-tools.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+<h2>Building lexical models in the IDE</h2>
+
+<p>
+    Double-click a <code><strong>.kpj</strong></code> file in Explorer to open the project in the
+    Keyman Developer Integrated Development Environment (IDE). You can also open a project from the Project menu within
+    the IDE. The project is automatically updated when you make changes, and changes specific to your computer are
+    saved in a <code><strong>.kpj.user</strong></code> file along with the <code><strong>.kpj</strong></code> file.
+    This <code><strong>.kpj.user</strong></code> file can be deleted at any time, and should not be saved to the
+    git repository.
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply developer/lexical-models/_shared/development-tools.php]: .gitignore already includes `*.kpj.user` and the `build/` folders (release, sample, experimental).

File: developer/lexical-models/index.php [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  Within each section of the folders, models are further grouped by the template <i>author</i> / <i>author.bcp47.uniq</i>.
+  For example, the folder structure may be:<br>
+  <b>release/nrc/nrc.en.mtnt/</b><br><br>
+  The components must be lower case and are:
+</p>
+<ul>
+  <li><b>author</b>: a short unique identifier, such as <i>nrc</i> or <i>sil</i></li>
+  <li><b>bcp47</b>: the canonical BCP 47 language tag for the model. For example <i>str</i> for SENĆOŦEN</li>
+  <li><b>uniq</b>: an optional component that can be provided when a language has multiple models from a single author.
+    For example, <i>example.en.custom</i> vs <i>example.en.wordlist</i>. We do recommend always using a uniquifier even if there are no
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply developer/lexical-models/index.php]: I'll rephrase **author** to be "a shortened/abbreviation" and describe **uniq** as "an optional unique identifier"

File: developer/lexical-models/submission/step-1.php [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+</pre>
+
+<h2 id="keep-fork-up-to-date">Adding the remote to the upstream lexical models repository</h2>
+
+<p>
+    From a command prompt, run the following:
+</p>
+
+<pre class="language-bash">
+<code>cd /d c:\projects\lexical-models\
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply developer/lexical-models/submission/step-1.php]: /facepalm

OK, to be consistent, I'll change them all to Unix-style (including keyboard submission guides)

File: developer/lexical-models/_shared/organize-folder-structure.php [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  <dt><code>extras\</code></dt>
+  <dd>other related files, e.g. scripts to format wordlists</dd>
+
+  <dt><code>source\</code></dt>
+  <dd>lexical model source files, including any files that will be packaged in the build</dd>
+
+  <dt><code>source\readme.htm</code></dt>
+  <dd>readme.htm and any resources referenced from the readme.htm file</dd>
+
+  <dt><code>source\welcome.htm</code></dt>
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply developer/lexical-models/_shared/organize-folder-structure.php]: This gets displayed during package installation. And it's available from the Model Information page.

File: developer/lexical-models/_shared/organize-folder-structure.php [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  <dt><code>build\</code></dt>
+  <dd>not stored in the repository; output files will go here</dd>
+
+  <dt><code>extras\</code></dt>
+  <dd>other related files, e.g. scripts to format wordlists</dd>
+
+  <dt><code>source\</code></dt>
+  <dd>lexical model source files, including any files that will be packaged in the build</dd>
+
+  <dt><code>source\readme.htm</code></dt>
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply developer/lexical-models/_shared/organize-folder-structure.php]: Starting  in Keyman 13, some of the Keyman apps will start displaying this before the user confirms package installation. Android app added this in keymanapp/keyman#2286 I don't know when iOS will update.

File: developer/lexical-models/submission/index.php [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  // Required
+  head([
+    'title' =>'Guide: Lexical Model Submissions'
+  ]);
+?>
+
+<h1>Guide for Lexical Model Submissions</h1>
+<p>
+  This is a guide for submitting lexical models to the lexical-models repository.
+  Lexical model developers will need to be proficient enough with Git to keep their own fork in sync with the upstream branch.
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply developer/lexical-models/submission/index.php]: I'll add some re-assuring statements about online resources being available. 

When Keyman Developer online rolls out, that will insulate users from dealing with this.

File: developer/lexical-models/submission/step-2.php [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+</pre>
+<br/>
+
+<p>
+   Get the latest changes from the master repository and push them to your fork with these following commands:
+</p>
+
+<pre class="language-bash">
+<code>cd /c/projects/lexical-models
+git checkout master
```
Comment (eddieantonio) [inline developer/lexical-models/submission/step-2.php]: No change required, but FYI, as of [git 2.23](https://github.blog/2019-08-16-highlights-from-git-2-23/#experimental-alternatives-for-git-checkout), the new command for this is `git switch`.

File: developer/lexical-models/index.php [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+  <li><b>uniq</b>:  an optional unique identifier that can be provided when a language has multiple models from a single author.
+    For example, <i>example.en.custom</i> vs <i>example.en.wordlist</i>. We do recommend always using a unique string even if there are no
+    current plans to produce more than one for a language.</li>
+</ul>
+
+<h2>Get Involved</h2>
+
+<ul class='menu'>
+    <p>
+        For users who are creating and submitting a lexical model, learn how to get set up with the lexical models
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply developer/lexical-models/index.php]: Yep. Adding a link to `"../12.0/guides/lexical-models/"`

File: developer/lexical-models/submission/step-2.php (hunk 11 above)
Comment (darcywong00) [reply developer/lexical-models/submission/step-2.php]: Nice! But maybe a little too experimental for my blood. 😨 
The **Description** to [git-switch](https://git-scm.com/docs/git-switch) ends with

> THIS COMMAND IS EXPERIMENTAL. THE BEHAVIOR MAY CHANGE.

**Pull request merged**